### ISSUE TITLE
URLSearchParams usage

### ISSUE THREAD

**Issue by niksy** [GoogleChromeLabs/native-url#7]
Do you plan on using native `URLSearchParams` API instead of `querystring` module to decode/encode query string? I see that it’s [mentioned in codebase](https://github.com/GoogleChromeLabs/native-url/blob/71705847b76a3ed020e29f8a95f4a575b51bb27e/src/parse.js#L143), but it’s not active since there is no Edge 16 support. Since this can also be polyfilled, maybe there is possibility to drop all 3rd party dependencies and go native all the way?

**Comment by janicklas-ralph** [comment]
The main reason for not using `URLSearchParams` is because it's not supported in Edge 16.
Since Edge 16 supports `type=module` scripts (our polyfilling technique)[https://github.com/GoogleChromeLabs/native-url#polyfill-for-older-browsers] would not work.
So I'm not sure how we can polyfill `URLSearchParams` for all `nomodule` browsers + Edge 16 without doing something more complicated like reading the user-agent.

Let me know your thoughts on this and if you think theres a better way to polyfill `URLSearchParams`

**Comment by niksy** [comment]
Yeah, I understand. I think it’s then better to wait for wider `URLSearchParams` browser support.

Nevertheless, I liked the idea of native-url module so I went and created [native-querystring](https://github.com/niksy/native-querystring), so that could work for developers who also want to polyfill `URLSearchParams` and use the convenience of `querystring` module!

Let me know if you’re interested in maybe mentioning this in your README (e.g. additional aliasing).
